beacon-chain/core/transition/deneb.go

```diff
@@ -0,0 +1,96 @@
+package transition
+
+import (
+	"crypto/sha256"
+	"encoding/binary"
+	"errors"
+	"fmt"
+)
+
+const (
+	blobCommitmentVersionKZG  uint8 = 0x01
+	blobTxType                      = 3
+	precompileInputLength           = 192
+	blobVersionedHashesOffset       = 258 // position of blob_versioned_hashes offset in a serialized blob tx, see TxPeekBlobVersionedHashes
+)
+
+// kzgToVersionedHash implements kzg_to_versioned_hash from EIP-4844
+func kzgToVersionedHash(kzg []byte) (h [32]byte) {
+	h = sha256.Sum256(kzg)
+	h[0] = blobCommitmentVersionKZG
+	return
+}
+
+// verifyKZGCommitmentsAgainstTransactions implements verify_kzg_commitments_against_transactions
+// from the EIP-4844 consensus spec:
+// https://github.com/ethereum/consensus-specs/blob/dev/specs/eip4844/beacon-chain.md#verify_kzg_commitments_against_transactions
+func verifyKZGCommitmentsAgainstTransactions(transactions, kzgCommitments [][]byte) error {
```

Comment:
we can remove these unused functions : )

Reply:
no, they have to be called from `core.transition`. These were called from Proto's library in devnet 5 but we removed those paths in devnet 6, will add those back soon

Reply:
you don't need these because EL is checking the version hashes now. You just need to convert the kzg commitments to version hashes and pass it to the EL. See: https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/beacon-chain.md#kzg_commitment_to_versioned_hash